sdk/tvl-usd.js

```diff
@@ -0,0 +1,138 @@
+const debug = require("debug")("opentvl:tvl-usd");
+const abi = require('./abis/aggregatorV3.json')
+const sdk = require('.');
+const fetch = require('node-fetch');
+const { getPriceFeeds } = require('./lib/chainlink');
+const wait = (milliseconds) => new Promise((resolve) => setTimeout(resolve, milliseconds));
+
+const COIN_GECKO_IDS = require('./data/coinGeckoIDs.json');
+
+let feeds = {};
+getPriceFeeds().then(res => {
+  debug("finished fetching price feed proxy addresses");
+  feeds = res;
+});
+
+function normalizeSymbol(symbol) {
+  const formatted = symbol.toUpperCase();
+  switch (formatted) {
+    case 'WBNB':
+    case 'BNB':
+      return 'BNB';
+    case 'WETH':
+    case 'ETH':
+      return 'ETH';
+    case 'WBTC':
+    case 'BTC':
+      return 'BTC';
+    case 'HDOT':
+    case 'DOT':
+      return 'DOT';
+    default:
+      return formatted;
+  }
+}
+
+function partitionTokens(tokenCounts) {
+  return Object.entries(tokenCounts).reduce((acc, [symbol, count]) => {
+    // The tokens in tokenCounts are derived from the tokenSet
+    // That token may be known by a different name in the feeds mapping
+    // Thus we need to call a normalize function to convert the token to the
+    // feeds naming system before we interact with the feeds
+    const normalized = normalizeSymbol(symbol);
+    const entry = { symbol: normalized, count };
+    if (feeds.eth[normalized]) {
+      acc.ethTokens.push(entry);
+    } else if (feeds.bsc[normalized]) {
+      acc.bscTokens.push(entry);
+    } else if (feeds.heco[normalized]) {
+      acc.hecoTokens.push(entry);
+    } else {
+      acc.coinGeckoTokens.push(entry);
+    }
+
+    return acc;
+  }, { ethTokens: [], bscTokens: [], hecoTokens: [], coinGeckoTokens: []})
+}
+
+async function convertToUSD(tokens, protocolFeeds, api) {
```

Comment:
can be moved to lib/chainlink.js

Reply:
this function is a good starting point for `sdk.eth.chainlink.getPrices([ { token0, token1 }, ...])`